pymc3/sampling.py

```diff
@@ -728,6 +742,10 @@ def _iter_sample(
                 point = step.step(point)
                 strace.record(point)
                 diverging = False
+            if callback is not None:
+                warns = getattr(step, "warnings", None)
+                callback(trace=strace, draw=Draw(chain, i == draws, i, i < tune, stats, point, warns))
```

Comment:
gross. we should really have single-threaded sampling return a `Draw`...

Reply:
That would be quite a bit nicer. I might have some time to work on another PR to do this